e2e/libs/recurringjob/rest.py

```diff
@@ -45,144 +47,142 @@ def add_to_volume(self, job_name, volume_name):
+                            return
+            except Exception as e:
+                logging(f"Iterating snapshot list error: {e}")
+            time.sleep(self.retry_interval)
+        assert False, f"since {current_time},\
+                        there's no new snapshot created by recurring job \
+                        {snapshot_list}"
+
+    def _check_backup_created(self, volume_name):
+        # check backup can be created by the recurring job
         current_time = datetime.utcnow()
         current_timestamp = current_time.timestamp()
-
+        logging(f"Recorded current time = {current_time}, timestamp = {current_timestamp}")
         label_selector=f"backup-volume={volume_name}"
-
-        max_iterations = period_in_sec * 10
-        for _ in range(max_iterations):
-            time.sleep(1)
-
+        backup_timestamp = 0
+        for i in range(self.retry_count):
+            logging(f"Waiting for {volume_name} new backup created ({i}) ...")
             backup_list = filter_cr("longhorn.io", "v1beta2", "longhorn-system", "backups", label_selector=label_selector)
-
-            if backup_list['items'] is None:
-                continue
-
-            for item in backup_list['items']:
-                backup_timestamp = datetime.strptime(item['metadata']['creationTimestamp'], '%Y-%m-%dT%H:%M:%SZ').timestamp()
-
-                if backup_timestamp > current_timestamp:
-                    return
-
-                logging(f"Backup {item['metadata']['name']} timestamp = {backup_timestamp} is not greater than {current_timestamp}")
-
-        assert False, f"No new backup created by recurringjob for {volume_name} since {current_time}"
+            try:
+                if len(backup_list['items']) > 0:
+                    for item in backup_list['items']:
+                        state = item['status']['state']
+                        if state != "Completed":
+                            continue
+                        backup_time = item['metadata']['creationTimestamp']
+                        backup_time = datetime.strptime(backup_time, '%Y-%m-%dT%H:%M:%SZ')
+                        backup_timestamp = backup_time.timestamp()
+                        if backup_timestamp > current_timestamp:
+                            logging(f"Got backup {item}, create time = {backup_time}, timestamp = {backup_timestamp}")
+                            return
+            except Exception as e:
+                logging(f"Iterating backup list error: {e}")
+            time.sleep(self.retry_interval)
+        assert False, f"since {current_time},\
+                        there's no new backup created by recurring job \
+                        {backup_list}"
 
     def cleanup(self, volume_names):
         for volume_name in volume_names:
-            logging(f"Cleaning up recurringjobs for volume {volume_name}")
+            logging(f"Cleaning up recurring jobs for volume {volume_name}")
```

Comment:
One thing needs to be mentioned is longhorn client itself uses naming like create_recurring_job‎ and by_id_recurring_job, even though this changes to single word, there is still inconsistency in the file.

Reply:
I think that it's fine since it's external to the longhorn-test repo. I understand there are some inconsistencies in the longhorn REST client. Let's just focus on the e2e test codebase now.

Reply:
Refined. Thanks!